### ISSUE TITLE
Always getting SaltCloudSystemExit on teardown of DigitalOcean VM

### ISSUE THREAD

**Issue by fake-name** [saltstack/salt#34295]
Basically, I'm creating and destroying a number of digitalocean VMs.

Every time I call `cc.destroy(clientname)` (where `cc = salt.cloud.CloudClient('/etc/salt/cloud')`, I get:

```
  File "salt_runner.py", line 164, in <module>
    herder.destroy_client("test-1")
  File "salt_runner.py", line 145, in destroy_client
    self.cc.destroy(clientname)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 371, in destroy
    mapper.destroy(names)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1032, in destroy
    ret = self.clouds[fun](name)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/digital_ocean.py", line 770, in destroy
    destroy_dns_records(name)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/digital_ocean.py", line 848, in destroy_dns_records
    response = query(method='domains', droplet_id=domain, command='records')
  File "/usr/lib/python2.7/dist-packages/salt/cloud/clouds/digital_ocean.py", line 542, in query
    request.text
SaltCloudSystemExit: An error occurred while querying DigitalOcean. HTTP Code: 404  Error: '{"id":"not_found","message":"The resource you were accessing could not be found."}'

```

This is because I'm not configuring any DNS records for the VM (it doesn't need any). As such, I have `create_dns_record: False` in the `cloud.providers` configuration for the Digital Ocean API.

Basically, it'd be nice to be able to tell if my call to `destroy()` actually worked. Note that all times I've tested it, *it did destroy my VM before the error was emitted*, so this isn't a huge issue, but it does clutter my logs up a fair bit.

I'm doing everything through the python scripting API. Sources are [here](https://github.com/fake-name/AutoTriever/blob/master/marshaller/salt_runner.py#L144).

### Steps to Reproduce Issue
 - Have a instantiated DO VM.
 - call `cc.destroy(<vm_name>)`

### Versions Report
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.14.32-xxxx-grs-ipv6-64
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by ahammond** [comment]
Duplicate of #33452 ?

**Comment by Ch3LL** [comment]
thanks for pointing out the duplicate @ahammond 

@fake-name yes this is a duplicate of #33452. I will close this issue and we can track it on that one. This is scheduled to get fixed for 2016.3.2 so this is on top of our list to get fixed.
